### PR TITLE
Removed sorting from `ListGroup` component

### LINKED ISSUE #311: Do not sort entries alphabetically
### Is your feature request related to a problem? Please describe.

With the properties panel rewrite we decide to make sorting of entries alphabetically the default. Since then we've unbuilt in the BPMN properties panel the sorting for most parts.

We did it because sorting alphabetically for anything where ordering (in XML) has an execution semantic is dangerous (https://github.com/camunda/camunda-modeler/issues/2932#issuecomment-1804223456). If at all, then users should have the chance to sort things manually.

Unfortunately the fact that sorting is the default continues to bite us, cf. https://github.com/camunda/camunda-modeler/issues/3978. I therefore propose to disable sorting per default, and instead understand that the properties panel operates in a domain where deterministic, user chosen sorting is the standard.

### Describe the solution you'd like

* Disable alphabetically sorting (make `shouldSort=false` per default)

### Describe alternatives you've considered

* Get rid of sorting by some label all together

### Additional context

Raised last (again) in the context of C7: https://github.com/camunda/camunda-modeler/issues/3978.

### PR DESCRIPTION
**Pull request by AlexanderSkrock**
This pull request aims to fix #311 . Currently, it is being developed.

- [x] Remove "shouldSort" property
- [x] Warn if this property is still used
- [x] Adjust testcases
- [x] Write a note in CHANGELOG.md
- [x] Discover dependant components in potentially reliant other projects. This includes to remove the flag and if sorting is necessary to reimplement the sorting on this level.

Closes #311

### CODE REVIEW

**Comment by nikku** [comment]
@AlexanderSkrock Great to see you picking this up. You should be able to link both projects, this core + bpmn-js-properties-panel to test them together:

```
cd bpmn-js-properties-panel
npm link ../properties-panel

...
```

**Comment by AlexanderSkrock** [comment]
Regarding necessary changes in the project `bpmn-js-properties-panel`, I saw the following.

**Camunda 7**
Usages without the `shouldSort` flags, thus using the previous default of sorting:
- FieldInjectionGroup
- InMappingGroup
- OutMappingGroup
- ExecutionListenerGroup
- TaskListenerGroup
- ConnectorInputGroup
- ConnectorOutputGroup
- ExtensionPropertiesGroup

Usages with the `shouldSort` flag set to `false`, thus already disabling sorting:
- ErrorsGroup
- FormDataGroup
- InputGroup
- OutputGroup
- ProcessVariablesGroup
- ExtensionPropertiesGroup

**Camunda 8**
Usages with the `shouldSort` flag set to `false`, thus already disabling sorting:
- ExtensionPropertiesGroup
- HeaderGroup
- InputGroup
- OutputGroup

**Changes to be made**
For all usages that already set the `shouldSort` flag set to `false`, we can simply remove the flag. For any other usage I will need to evalute the necessity of sorting.

**Comment by AlexanderSkrock** [comment]
This pull request together with changes made in [#1047 in bpmn-properties-panel](https://github.com/bpmn-io/bpmn-js-properties-panel/pull/1047) should be good to go now, from my point of view.

One more question, I saw that we released `properties-panel` version 3.18.2 already, so we would need to increate to 3.18.2 and also update it for `bpmn-js-properties-panel`. Is this a step which is done automatically during the release process?

**Comment by nikku** [comment]
Once you did it then I can give it my review.

**Comment by AlexanderSkrock** [comment]
> @AlexanderSkrock You'd want to rebase this PR on main before merge.

I checked once again, but it seems my commits were based on main's head already.

**Comment by nikku** [comment]
Will look into your changes today. Thanks already for your work :clap:

**Comment by nikku** [comment]
After some distractions I hope I find the time to finalize my review + testing tomorrow.

**Comment by nikku** [comment]
I found the time to review the change, along with https://github.com/bpmn-io/bpmn-js-properties-panel/pull/1047.

A few notes:

* From what it looks to me the change is incomplete. Looking at sorting at the core [`List#compareFn`](https://github.com/bpmn-io/properties-panel/blob/1b2cbec8e55bfb470695bde2a302e60110a65261/src/components/entries/List.js#L54) is used along with sorting to implement sort features; without sorting there is no necessity for `compareFn`, we want to ditch it which I sketched in https://github.com/bpmn-io/bpmn-js-properties-panel/pull/1047/commits/d883a0fcc3cb19ddf422b1ab3a86bc1595cfb28a.
* Integration testing https://github.com/bpmn-io/bpmn-js-properties-panel/pull/1047 I found a bunch of places where auto focus is now broken; we want to ensure that items are added last to the list (no magic), and focused regardless, sketched for `Execution Listeners > Execution Listener > Field injections` via https://github.com/bpmn-io/bpmn-js-properties-panel/pull/1047/commits/99535149a0297c369372666cda6895f1411707f2). Without that change in the relevant places we see the following interaction, shown on the example of `User Task (Generated Form): Form Fields > Field > Properties`:
  
  capture nSTWvz_optimized

Other than that the solution feels solid, improves C7, and does not break C8 functionality (where we do not sort already :tm:). Good job!

**Review by nikku — CHANGES_REQUESTED** [review]
Consider to ditch `List#compareFn`, cf. https://github.com/bpmn-io/properties-panel/pull/353#issuecomment-2096326398.

**Review by nikku — APPROVED** [review]

**Comment by nikku** [comment]
Waiting for CI to pass before we go ahead and merge this.

File: CHANGELOG.md
Comment (nikku) [inline CHANGELOG.md:9]:
```suggestion
* `FEAT`: do not sort list items alphabetically ([#311](https://github.com/bpmn-io/properties-panel/issues/311) 
* `CHORE`: remove sorting for list groups ([#353](https://github.com/bpmn-io/properties-panel/pull/353))
```

**Pull request merged**
